src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -2226,77 +2238,173 @@ internal void HeartBeat()
                 memberName,                  // method name
                 message);                    // message
 
+        private struct RequestQueueItem<T>
+        {
+            public HttpRequestMessage Request;
+            public TaskCompletionSourceWithCancellation<T> Waiter;
+        }
```

Comment:
Nit: consider making this a nested type in `RequestQueue`, e.g.
```C#
private struct RequestQueue<T>
{
    ...

    public struct Entry
    {
        public HttpRequestMessage Request;
        public TaskCompletionSourceWithCancellation<T> Waiter;
    }
}
```
That better aligns with other types like `Dictionary<>`, means that the implementation of the queue only needs to say `Entry` rather than `RequestQueueItem<T>`, etc.